File: python/euler2.py
def main(lim):
	res = 0
	prev, n = 0, 1
	while n <= lim:
		# print(n)
		if n%2 == 0:
			res += n
		prev, n = n, prev+n
	return res


if __name__ == '__main__':
	print(main(4000000))

File: javascript/euler2.js
function sum_of_even_fibs(lim)
{
    var res = 0
    var prev = 1
    var n = 1
    var tmp
    while(n <= lim)
    {
        if (n%2 == 0)
        {
            res += n
        }
        tmp = prev
        prev = n
        n = n + tmp
    }
    return res
}

console.log(sum_of_even_fibs(4000000))
